tuist/tuist
# Respect overriding the TEST_HOST build setting

…, even when using xcconfig.

Resolves <https://github.com/tuist/tuist/issues/6888>

### Short description 📝

 Fix Tuist overrides the `TEST_HOST` build setting for UnitTest targets, even when using xcconfig.


**Concerns**:
> If this issue fix is implemented, one concern is that users who have set defaultSettings to `.none` and have not explicitly provided a `TEST_HOST` in the code will face build failures in their projects.

> Personally, considering the implications of the comment regarding `defaultSettings`, it seems inconsistent to have code that overrides `TEST_HOST` with specific properties of the target, even when `defaultSettings` is set to `.none`. 
If it is `.none`, I believe users should have the right to customize their settings fully in accordance with that meaning.

>(I have personally experienced instances where I forgot the values I had previously set for properties despite using `.none`, resulting in additional debugging time.)

### How to test the changes locally 🧐


1. Write test code in ConfigGeneratorTests to ensure that when defaultSettings is either .`none` or `.recommended` and `.essential`, if `TEST_HOST` is included in `excluding`, then `TEST_HOST` should be nil.
2.  Generate the attached project with Tuist and check the `TEST_HOST` in the build settings of the unit test target (xcconfig attached).
[unittest.zip](https://github.com/user-attachments/files/17422586/unittest.zip)


### Contributor checklist ✅


### Reviewer checklist ✅
